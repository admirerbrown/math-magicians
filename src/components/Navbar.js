import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <div className="header">
    <h1 className="banner">Math Magicians</h1>
    <nav className="nav-link">
      <Link to="/" className="nav-item">
        Home
        <span className="divider">|</span>
      </Link>
      <Link to="/calculator" className="nav-item">
        Calculator
        <span className="divider">|</span>
      </Link>
      <Link to="/quote" className="nav-item">
        Quote
      </Link>
    </nav>
  </div>
);
export default Navbar;
