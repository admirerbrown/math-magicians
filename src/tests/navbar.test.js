import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Navbar from '../components/Navbar';

describe('Test the navbar component', () => {
  it('has all snapshots working', () => {
    const tree = renderer.create(<Navbar />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    render(<Navbar />);
  });
});
