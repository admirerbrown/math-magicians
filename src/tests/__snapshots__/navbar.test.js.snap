// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the navbar component has all snapshots working 1`] = `
<div
  className="header"
>
  <h1
    className="banner"
  >
    Math Magicians
  </h1>
  <nav
    className="nav-link"
  >
    <a
      className="nav-item"
      href="/"
      onClick={[Function]}
    >
      Home
      <span
        className="divider"
      >
        |
      </span>
    </a>
    <a
      className="nav-item"
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
      <span
        className="divider"
      >
        |
      </span>
    </a>
    <a
      className="nav-item"
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </nav>
</div>
`;
